src/utils/user-agent-utils.ts

```diff
@@ -0,0 +1,366 @@
+import { _includes } from './index'
+import { _isUndefined } from './type-utils'
+
+/**
+ * this device detection code is (at time of writing) about 3% of the size of the entire library
+ * this is mostly because the identifiers user in regexes and results can't be minified away since
+ * they have meaning
+ *
+ * so, there are some pre-uglifying choices in the code to help reduce the size
+ * e.g. many repeated strings are stored as variables and then old-fashioned concatenated together
+ *
+ * TL;DR here be dragons
```

Comment:
3% of the library for some regexes and some strings is wild. it's just not very minifiable because the literal strings have meaning

so, we can do silly things

removing duplication - by using lots of constants primarily - will help drive this down

but it's also overly complicated and so its hard to see where the duplication is necessary